scikit_mol/utilities.py

```diff
@@ -53,3 +53,18 @@ def sanitize(self, X_smiles_list, y=None):
             self.errors = pd.DataFrame({'SMILES':X_errors})
 
             return X_out, X_errors
+
+
+def get_mols_from_X(X):
```

Comment:
This is not the place to do this I think. We should not make these kinds of column selections at this level. I would say, that we need to fix the input and output of smilesconverter and molfromsmiles converter to accept 2D input and pandas dataframes, under the assumption that there's only one column. We can assert that and throw a sensible error message about using ColumnTransformer if picking it out of a dataframe. Currently they accept iterables and returns lists or flat arrays, which isn't the way according to sklearn.
Although not completely in the vane of sklearn, I would like to keep the option to input lists and flat arrays as it's so fast to convert a list of smiles or mols with the tools.

Reply:
Thinking of it, this lack of column vector or pandas input support is likely contributing to crashing your attempts to use column transformer.

Reply:
> Thinking of it, this lack of column vector or pandas input support is likely contributing to crashing your attempts to use column transformer.

Yes, I also thought that if we change how scikit-mol handles transformer input in this PR (by making it more compatible with scikit-learn), ColumnTransformer is more likely to work. I will try after this PR is ready!